zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -423,15 +425,26 @@ public void operationComplete(Future<Channel> future) {
 
                     String authProviderProp = System.getProperty(x509Util.getSslAuthProviderProperty(), "x509");
 
-                    X509AuthenticationProvider authProvider = (X509AuthenticationProvider) ProviderRegistry.getProvider(authProviderProp);
+                    X509AuthenticationProvider authProvider = null;
+                    ServerAuthenticationProvider serverAuthProvider = null;
```

Comment:
Could we add a descriptive comment why this is necessary here? I think we should give enough context so anyone who sees this can understand the reason behind this change :)